### PR TITLE
Add support for Utf8View in arrow_string::length

### LINKED ISSUE #6305: Add support for Utf8View in arrow_string::length
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

This came up as a result of a bug report in DataFusion: https://github.com/apache/datafusion/issues/12149 where `length(StringViewArray)` throws an error.

**Describe the solution you'd like**

The length function has StringViewArray support added and tests added to cover this case.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Omega359**
# Which issue does this PR close?

Closes #6305

# Rationale for this change
 

Add support for StringViewArray in length function

# What changes are included in this PR?

Code and tests.

# Are there any user-facing changes?

No

### CODE REVIEW

File: arrow-string/src/length.rs [hunk 1]
```diff
@@ -74,6 +74,14 @@ pub fn length(array: &dyn Array) -> Result<ArrayRef, ArrowError> {
...
+            let list = array.as_string_view();
+            let v = list.views().iter().map(|v| *v as i32).collect::<Vec<_>>();
+            Ok(Arc::new(PrimitiveArray::<Int32Type>::new(
+                v.into(),
+                list.nulls().cloned(),
+            )))
+        }
         DataType::Binary => {
             let list = array.as_binary::<i32>();
             Ok(length_impl::<Int32Type>(list.offsets(), list.nulls()))
```
Comment (XiangpengHao) [inline arrow-string/src/length.rs:87]: We should also consider adding support for `BinaryView` as well, maybe in a follow-up PR

Comment (Omega359) [reply arrow-string/src/length.rs:87]: I can do that, sure.

Comment (Omega359) [reply arrow-string/src/length.rs:87]: https://github.com/apache/arrow-rs/issues/6358

File: arrow-string/src/length.rs [hunk 2]
```diff
@@ -147,9 +155,15 @@ mod tests {
 
     fn length_cases_string() -> Vec<(Vec<&'static str>, usize, Vec<i32>)> {
         // a large array
-        let values = ["one", "on", "o", ""];
+        let values = [
+            "one",
+            "on",
+            "o",
+            "",
+            "this is a longer string to test string array with",
```
Comment (alamb) [inline arrow-string/src/length.rs:163]: ❤️

**Review by alamb — APPROVED** [review]
Thanks @Omega359  and @XiangpengHao  🚀

**Pull request merged**